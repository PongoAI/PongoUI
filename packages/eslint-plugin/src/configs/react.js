/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
