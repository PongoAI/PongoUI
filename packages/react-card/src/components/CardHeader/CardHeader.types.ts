import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type CardHeaderSlots = {
  /**
   * The root of CardHeader.
   */
  root: Slot<'div'>;

  /**
   * The avatar or image indicator for the Header.
   */
  avatar?: Slot<'div'>;

  /**
   * The text content wrapper for the CardHeader.
   */
  textContainer?: Slot<'div'>;

  /**
   * The header text for the CardHeader.
   */
  header?: Slot<'span'>;

  /**
   * The description for the CardHeader.
   */
  description?: Slot<'span'>;

  /**
   * The action item/s for the CardHeader.
   */
  action?: Slot<'div'>;
};

export type CardHeaderProps = ComponentProps<CardHeaderSlots>;

export type CardHeaderState = ComponentState<CardHeaderSlots>;
