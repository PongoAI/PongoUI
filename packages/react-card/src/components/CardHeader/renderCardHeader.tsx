import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { CardHeaderState, CardHeaderSlots } from './CardHeader.types';

export const renderCardHeader = (state: CardHeaderState) => {
  const { slots, slotProps } = getSlots<CardHeaderSlots>(state);

  return (
    <slots.root {...slotProps.root}>
      {slots.avatar && <slots.avatar {...slotProps.avatar} />}
      {slots.textContainer && (slots.header || slots.description) && (
        <slots.textContainer {...slotProps.textContainer}>
          {slots.header && <slots.header {...slotProps.header} />}
          {slots.description && <slots.description {...slotProps.description} />}
        </slots.textContainer>
      )}
      {slots.action && <slots.action {...slotProps.action} />}
    </slots.root>
  );
};
