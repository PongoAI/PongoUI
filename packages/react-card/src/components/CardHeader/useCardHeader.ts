import * as React from 'react';
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
import type { CardHeaderProps, CardHeaderState } from './CardHeader.types';

export const useCardHeader = (props: CardHeaderProps, ref: React.Ref<HTMLElement>): CardHeaderState => {
  const { avatar, textContainer, header, description, action } = props;

  const state: CardHeaderState = {
    components: {
      root: 'div',
      avatar: 'div',
      textContainer: 'div',
      header: React.Fragment,
      description: React.Fragment,
      action: 'div',
    },
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),
    avatar: resolveShorthand(avatar, {
      required: false,
    }),
    textContainer: resolveShorthand(textContainer || {}),
    header: resolveShorthand(header, {
      required: false,
    }),
    description: resolveShorthand(description, {
      required: false,
    }),
    action: resolveShorthand(action),
  };

  return state;
};
