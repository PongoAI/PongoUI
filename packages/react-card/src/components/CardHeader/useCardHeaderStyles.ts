import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { CardHeaderState } from './CardHeader.types';
import { tokens } from '@pongo-ui/react-theme';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    flexShrink: 0,
    ...shorthands.gap(tokens.focusedGap),
  },

  avatar: {
    display: 'flex',
    minWidth: '24px',
    minHeight: '24px',
    maxWidth: '32px',
    maxHeight: '32px',
    '> *': {
      minWidth: 'inherit',
      minHeight: 'inherit',
      maxWidth: '100%',
      maxHeight: '100%',
    },
  },

  textContainer: {
    display: 'flex',
    flexDirection: 'column',
    flexGrow: 1,
    height: 'inherit',
    justifyContent: 'center',
  },
});

export const useCardHeaderStyles = (state: CardHeaderState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(styles.root, state.root.className);

  if (state.textContainer) {
    state.textContainer.className = mergeClasses(styles.textContainer, state.textContainer.className);
  }

  if (state.avatar) {
    state.avatar.className = mergeClasses(styles.avatar, state.avatar.className);
  }

  return state;
};
