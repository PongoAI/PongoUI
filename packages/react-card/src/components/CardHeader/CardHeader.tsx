import * as React from 'react';
import { useCardHeader } from './useCardHeader';
import { renderCardHeader } from './renderCardHeader';
import { useCardHeaderStyles } from './useCardHeaderStyles';
import type { CardHeaderProps } from './CardHeader.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The CardHeader provides information about the Card components use.
 */
export const CardHeader: ForwardRefComponent<CardHeaderProps> = React.forwardRef((props, ref) => {
  const state = useCardHeader(props, ref);

  useCardHeaderStyles(state);

  return renderCardHeader(state);
});
CardHeader.displayName = 'CardHeader';
