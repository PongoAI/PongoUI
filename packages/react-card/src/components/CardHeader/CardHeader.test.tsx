import * as React from 'react';
import { CardHeader } from './CardHeader';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('CardHeader', () => {
  isConformant({
    Component: CardHeader,
    displayName: 'CardHeader',
  });

  it('renders a basic CardHeader', () => {
    const renderedComponent = render(
      <CardHeader avatar={<div>CZ</div>} header={'This is a header'} description={'This is a subHeadline'} />,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
