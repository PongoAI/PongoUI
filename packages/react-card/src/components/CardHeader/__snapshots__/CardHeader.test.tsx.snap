// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders a basic CardHeader 1`] = `
<body>
  <div>
    <div
      class="___100u5yg f22iagw f1063pyq f122n59 f4d9j23 fi64zpg f1kbxzs8 f1nke1mu"
    >
      <div
        class="___1y0vsyi f22iagw f17fgpbq f1pha7fy f44c6la f17bzqcb fgn7ydb f1jevvm1 fqihwjc f6x1sus"
      >
        <div>
          CZ
        </div>
      </div>
      <div
        class="___blydbu0 f22iagw f1vx9l62 fqerorx fwk23hs f4d9j23"
      >
        This is a header
        This is a subHeadline
      </div>
    </div>
  </div>
</body>
`;
